src/components/home/ProductList/index.tsx

```diff
@@ -1,17 +1,42 @@
 import type { ReactElement } from 'react'
+import { useEffect, useState } from 'react'
+import { useInView } from 'react-intersection-observer'
 import { Styled } from './styled'
 import type { ProductListProps } from './types'
 import { ProductItem } from '../ProductItem'
 
-const ProductList = ({ productList }: ProductListProps): ReactElement => {
+const ProductList = ({
+  postData,
+  hasNextPage,
+  fetchNextPage
+}: ProductListProps): ReactElement => {
+  const [isFirstRender, setIsFirstRender] = useState<boolean>(false)
+
+  const { ref: isLastPrdRef, inView } = useInView({
+    threshold: 1
+  })
+
+  useEffect(() => {
+    if (inView && hasNextPage) {
+      if (!isFirstRender) {
+        setIsFirstRender(true)
+        return
+      }
+      fetchNextPage && fetchNextPage()
+    }
+  }, [inView, hasNextPage, fetchNextPage])
+
   return (
     <>
       <Styled.NewProductTitle>새로운 상품</Styled.NewProductTitle>
       <Styled.ProductListWrapper>
-        {productList.map(item => (
-          <ProductItem key={item.id} productItem={item} />
-        ))}
+        {postData?.pages?.map(page =>
```

Comment:
```suggestion
        {pages?.map(page =>
```
내부에서 pages만 필요하다면 상위에서 postData.pages로 보내줘도 괜찮을 것 같아요!

Reply:
[완료](https://github.com/price-offer/offer-fe/pull/168/commits/02a1412339ae0a5f5e5ad6532c83d618a429873e#diff-18e0d4553c97cfc420e938ccafb4e3a688e782fa4512ba4ceae3e2a6f24c1987)